### PR TITLE
feat(PPDSC-2416): update newskit and packages versions

### PR DESCRIPTION
**Pull request by evgenitsn**
- Updated newskit to v6.0.0 and nextjs to v12.3.0
- Fixed breaking changes
- Updated demo code blocks

Full list of updated packages:

    "@emotion-icons/material": "^3.14.0",
    "@emotion/react": "^11.10.4",
    "@emotion/styled": "^11.10.4",
    "@types/react": "^18.0.0",
    "@types/react-dom": "^18.0.0",
    "newskit": "6.0.0",
    "next": "12.3.0",
    "react": "18.2.0",
    "react-dom": "18.2.0",
    "react-syntax-highlighter": "^13.0.0",

### CODE REVIEW

File: components/template.tsx [hunk 1]
```diff
@@ -212,15 +214,15 @@ export default ({ title = 'NewsKit Demo', children }) => (
...
-        
+
         body {
           margin: 0;
           padding: 0;
         }
       `}
     />
-    <ThemeProvider theme={theme}>
+    <NewsKitProvider theme={theme}>
```
Comment (mutebg) [inline components/template.tsx:224]: I am not sure if you need this one, in the whole app you must ave only 1 newskit provider and from my understanding, there is already one coming from `useThemeSwitcher`.

File: components/code-template.tsx [hunk 2]
```diff
@@ -71,22 +75,33 @@ export default ({
...
-              Next Step 👉
-            </Tag>
-          ) : (
-            <Tag href="/" size="large">
-              Home 🏡
-            </Tag>
-          )}
+          <Block>
+            {prevPage && (
+              <Tag
```
Comment (mutebg) [inline components/code-template.tsx:87]: Can you please add next-link here so transitions between pages is smooth,  something like 
```
<Link href=""><Tag ... /></Link>
```

File: public/static/page6.tsx [hunk 3]
```diff
@@ -25,36 +28,38 @@ const NoWrapButton = styled(Button)`
 `;
```
Comment (mutebg) [inline public/static/page6.tsx:28]: Now `white-space: nowrap` can be set as style preset, wondering if we can address that here.
Edit: feel free to ignore it, maybe is better for another ticket in which we change some of the code

File: components/use-theme-switcher.tsx [hunk 4]
```diff
@@ -61,15 +69,28 @@ const StyledSelect = styled.select`
...
+    'textInput.medium.input',
+    'input',
+    'spaceInset'
+  )}
+  ${getResponsiveSize(
+    'minHeight',
+    'textInput.medium.input',
+    'input',
+    'minHeight'
+  )}
```
[the remark this answers is not in the document]
Comment (evgenitsn) [reply components/use-theme-switcher.tsx]: Updated

File: components/template.tsx (hunk 1 above)
Comment (evgenitsn) [reply components/template.tsx]: Updated

File: components/code-template.tsx (hunk 2 above)
Comment (evgenitsn) [reply components/code-template.tsx]: Updated

File: public/static/page6.tsx (hunk 3 above)
Comment (evgenitsn) [reply public/static/page6.tsx:28]: I will skip this one for now.

**Review by mutebg — APPROVED** [review]

**Review by Xin00163 — APPROVED** [review]

**Pull request merged**